src/overlays/actors/ovl_En_Niw/z_en_niw.c

```diff
@@ -21,52 +40,1012 @@ const ActorInit En_Niw_InitVars = {
+                this->unkTimer252 = this->unkTimer250 = this->unk298;
+
+                this->unk264[8] = 0.0f;
+                this->unk264[6] = 0.0f;
+                this->unk264[5] = 0.0f;
+                this->unk264[7] = 0.0f;
+                this->niwStormActive = true;
+                this->actionFunc = EnNiw_Trigger;
+                this->unk304 = 0.0f;
+                this->unk300 = 0.0f;
+                this->actor.speedXZ = 0.0f;
+
+            } else {
+                this->unkTimer260 = 10;
+                this->sfxTimer1 = 30;
+                this->unk29E = 1;
+                Audio_PlayActorSound2(&this->actor, NA_SE_EV_CHICKEN_CRY_M); // crow
+                this->unkTimer254 = 100;
+                this->unk2EC = 0;
+                EnNiw_SetupRunning(this);
+            }
+        }
+    }
+}
+
+void EnNiw_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnNiw* this = (EnNiw*)thisx;
+    s8 pad0;
+    s16 i;
+    ActorPlayer* player = PLAYER;
+    s16 pad1;
+    s16 featherCount;
+    Vec3f pos;
+    Vec3f spB8;
+    Vec3f spAC;
+    s32 pad2[9];
+    s16 temp29C;
+    f32 featherScale;
+    f32 camResult;
+    f32 floorHeight;
+    f32 dist = 20.0f;
+    s32 pad3;
+
+    this->unusedCounter28C++;
+
+    if (this->niwType == ENNIW_TYPE_UNK1) {
+        this->actor.shape.rot.y = this->actor.world.rot.y = this->actor.parent->shape.rot.y;
+    }
+
+    if (this->unknownState28E != 0) {
+        this->unk264[9] = 0.0f;
+    }
+
+    if (this->unk29E != 0) {
+        featherCount = 20;
+        if (this->unk29E == 2) {
+            featherCount = 4;
+        }
+        for (i = 0; i < featherCount; ++i) {
+            pos.x = randPlusMinusPoint5Scaled(10.0f) + this->actor.world.pos.x;
+            pos.y = randPlusMinusPoint5Scaled(10.0f) + (this->actor.world.pos.y + this->unk308);
+            pos.z = randPlusMinusPoint5Scaled(10.0f) + this->actor.world.pos.z;
+            featherScale = Rand_ZeroFloat(6.0f) + 6.0f;
+
+            if ((this->unk29E == 2) && (this->unk308 != 0)) {
+                pos.y += 10.0f;
+            }
+
+            if (this->unk308 == 0) {
+                featherScale = Rand_ZeroFloat(2.0f) + 2.0f;
+            }
+            spB8.x = randPlusMinusPoint5Scaled(3.0f);
+            spB8.y = (Rand_ZeroFloat(2.0f) * 0.5f) + 2.0f;
+            spB8.z = randPlusMinusPoint5Scaled(3.0f);
+            spAC.z = spAC.x = 0.0f;
+            spAC.y = -0.15000000596f;
+
+            EnNiw_SpawnFeather(this, &pos, &spB8, &spAC, featherScale);
+        }
+        this->unk29E = 0;
+    }
+
+    EnNiw_UpdateFeather(this, globalCtx);
+
+    DECR(this->unkTimer24C);
+    DECR(this->unkTimer24E);
+    DECR(this->unkTimer250);
+    DECR(this->unkTimer252);
+    DECR(this->unkTimer254);
+    DECR(this->sfxTimer1);
+    DECR(this->flutterSfxTimer);
+    DECR(this->unusedTimer25A);
+    DECR(this->yawTimer);
+    DECR(this->unusedTimer25E);
+    DECR(this->unkTimer260);
```

Comment:
Are all these nothing more than `this->timer--`, if so I think that'd be better as `DECR` is more for if-statements:
https://discord.com/channels/688807550715560050/688851997352263820/844406587870937119

Reply:
No these do the full DECR macro, checking if the value is zero first before reducing by one
https://github.com/isghj5/mm/blob/d124db97ba9743f36761215fa7c44f9126a021bd/src/overlays/actors/ovl_En_Niw/z_en_niw.c#L814-L846

Reply:
I personally think it looks better as it was with the explicit if-statements, since `DECR` and ternaries are intended to return values, but I'll leave it up to you

Reply:
At least in OoT we don't use the `DECR` macro outside of if conditionals. I don't what's the convention in MM. I think this should be discussed in the discord server.